fido/src/main/java/com/yubico/yubikit/fido/ctap/FingerprintBioEnrollment.java

```diff
@@ -0,0 +1,444 @@
+        @Nullable
+        private final Integer timeout;
+        @Nullable
+        private byte[] templateId;
+        @Nullable
+        private Integer remaining;
+
+        public Context(
+                FingerprintBioEnrollment bioEnrollment,
+                @Nullable Integer timeout,
+                @Nullable byte[] templateId,
+                @Nullable Integer remaining) {
+            this.bioEnrollment = bioEnrollment;
+            this.timeout = timeout;
+            this.templateId = templateId;
+            this.remaining = remaining;
+        }
+
+        /**
+         * Capture a fingerprint sample.
+         * <p>
+         * This call will block for up to timeout milliseconds (or indefinitely, if
+         * timeout not specified) waiting for the user to scan their fingerprint to
+         * collect one sample.
+         *
+         * @return None, if more samples are needed, or the template ID if enrollment is
+         * completed.
+         * @throws IOException      A communication error in the transport layer.
+         * @throws CommandException A communication in the protocol layer.
+         * @throws CaptureError     An error during fingerprint capture.
+         */
+        @Nullable
+        public byte[] capture() throws IOException, CommandException, CaptureError {
+            int sampleStatus;
+            if (templateId == null) {
+                final EnrollBeginStatus status = bioEnrollment.enrollBegin(timeout);
+                templateId = status.getTemplateId();
+                remaining = status.getRemaining();
+                sampleStatus = status.getSampleStatus();
+            } else {
+                final CaptureStatus status = bioEnrollment.enrollCaptureNext(templateId, timeout);
+                remaining = status.getRemaining();
+                sampleStatus = status.getSampleStatus();
+            }
+
+            if (sampleStatus != FEEDBACK_FP_GOOD) {
+                throw new CaptureError(sampleStatus);
+            }
+
+            if (remaining == 0) {
+                return templateId;
+            }
+
+            return null;
+        }
+
+        /**
+         * Cancels ongoing enrollment.
+         *
+         * @throws IOException      A communication error in the transport layer.
+         * @throws CommandException A communication in the protocol layer.
+         */
+        public void cancel() throws IOException, CommandException {
+            bioEnrollment.enrollCancel();
+            templateId = null;
+        }
+    }
+
+    public FingerprintBioEnrollment(
+            Ctap2Session ctap,
+            PinUvAuthProtocol pinUvAuthProtocol,
+            byte[] pinUvToken) throws IOException, CommandException {
+        super(ctap, BioEnrollment.MODALITY_FINGERPRINT);
+        this.pinUvAuth = pinUvAuthProtocol;
+        this.pinUvToken = pinUvToken;
+        this.cachedSensorInfo = readFingerprintSensorInfo();
```

Comment:
This instance method call in the constructor rings the same kind of warning bell, but it's probably not worth refactoring this and the `call` methods since those are also needed as instance methods after the instance has been constructed.

Reply:
Fixed this 👍🏻